### ISSUE TITLE
Long model names break db:schema:load on Postgres

### ISSUE THREAD

**Issue by skiningham** [rails/rails#28332]
### Steps to reproduce
1. `$ rails new issue_reproduction --database=postgresql`
1. `$ rails g scaffold MyVeryLongExampleModelNameBecauseIThinkSchemaLoadBreaksSometimes title:string body:text`
1. `$ rails db:create db:migrate`
1. `$ rails db:drop db:create db:schema:load`

Example repo: https://github.com/skiningham/issue_reproduction
Single-file reproduction: https://gist.github.com/skiningham/fb6f380c75ca5a6c65909958f5f52a31

### Expected behavior
`$ rails db:schema:load` should succeed without errors.

### Actual behavior
The command fails with:
```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "my_very_long_example_model_name_because_i_think_schema_l_id_seq" does not exist
: CREATE TABLE "my_very_long_example_model_name_because_i_think_schema_load_bre" ("id" bigint DEFAULT nextval('my_very_long_example_model_name_because_i_think_schema_l_id_seq'::regclass) NOT NULL PRIMARY KEY, "title" character varying, "body" text, "created_at" timestamp NOT NULL, "updated_at" timestamp NOT NULL)
```

### System configuration
**Rails version**: 5.0.2, 5.1.0.beta1

**Ruby version**: 2.2.3p173

**Postgres version**: 9.5.4

### Additional information
I can reproduce this error with ActiveRecord 5.0.2 and 5.1.0.beta1. I did not try against master. I cannot reproduce this error against ActiveRecord 4.2.8. Apologies for the rough single-file reproduction; this error does not occur against SQLite so I made an attempt at adjusting the default template for Postgres. `$ rails db:structure:load` succeeds in the example application when a `structure.sql` is dumped. `$ rails db:drop db:create db:migrate` succeeds in the example application.

**Comment by pixeltrix** [comment]
This is because our method for detecting whether the primary key column is a serial assumes that the name hasn't been truncated:

``` ruby
def serial?
  return unless default_function

  %r{\Anextval\('"?#{table_name}_#{name}_seq"?'::regclass\)\z} === default_function
end
```

This results in the horrible dump you get in your schema.rb:

``` ruby
create_table "my_very_long_example_model_name_because_i_think_schema_load_bre", id: :integer, default: -> { "nextval('my_very_long_example_model_name_because_i_think_schema_l_id_seq'::regclass)" }, force: :cascade do |t|
```

Unfortunately at that point the `serial?` method doesn't have access to the connection object so we'll need to fix it at a higher level, probably in `column_spec_for_primary_key` which is on the connection.

We can use `pg_depend` to find the sequence that's dependent on the primary key and then adjust the attributes where we override `column_spec_for_primary_key` in the PostgreSQL adapter.

**Comment by kamipo** [comment]
Fixed in 8877492df2853cba92d95dabdfcdee7244e7d8f0. Backported in af9c1707ad49eb421cf76c1367da3efbf3c39aea and 7025b1d8ebf694c288c71878b5229394ba3c0580.
